client/blocks/importer/hooks/use-site-eligible-migration-trial-plan.ts

```diff
@@ -0,0 +1,15 @@
+import { isEnabled } from '@automattic/calypso-config';
+import { useSelector } from 'calypso/state';
+import { getCurrentUser } from 'calypso/state/current-user/selectors';
+
+/**
+ * Allow only one migration trial site per user
+ * - a8c team members are allowed to have more than one.
+ * - user with a flag are allowed to have more than one.
+ */
+export function useSiteEligibleMigrationTrialPlan(): boolean {
+	const user = useSelector( getCurrentUser );
+	const flags = user?.meta?.data?.flags?.active_flags || [];
+
+	return isEnabled( 'plans/migration-trial' ) && flags.includes( 'allow_multiple_business_trials' );
+}
```

Comment:
Sorry for the trouble @bogiii . I know we spoke about this and I was thinking in the wrong direction yesterday, because we miss one case here, if a user had a migration trial before on other sites then this check won't work. I've created a new endpoint to check if the trial plan is eligible or not now. See D118425-code

So you can consume it and check if `eligible` is `true` here. All other checking logic is handled in this new endpoint as well so you don't need to check the feature flags either. Please let me know if you have any questions, thank you 🙏